cmd/config/etcd/dns/etcd_dns.go

```diff
@@ -154,6 +154,8 @@ func (c *CoreDNS) list(key string) ([]SrvRecord, error) {
 
 // Put - Adds DNS entries into etcd endpoint in CoreDNS etcd message format.
 func (c *CoreDNS) Put(bucket string) error {
+	c.Delete(bucket) // delete any existing entries.
```

Comment:
Unclear on why we need to delete it first..?

Reply:
Because we need to update the previous entries, they may be stale @nitisht